src/plugins/intel_gna/src/gna_transformations_pipeline.cpp

```diff
@@ -72,56 +77,112 @@ void TransformationsPipeline::apply(const std::shared_ptr<ov::Model>& model,
 
     // In OV API 2.0(IRv10) default convertion to fp32 (inputs, outputs and weights) is disabled
     // and we need to run the ConvertPrecision transformation to support old networks.
-    manager.register_pass<ov::pass::ConvertPrecision>(precisions_map{{ngraph::element::f16, ngraph::element::f32}});
-    manager.register_pass<ov::pass::ConvertMVN1ToMVN6>();
-    manager.register_pass<ov::intel_gna::pass::DecomposeMVN>();
-    manager.register_pass<ov::pass::CommonOptimizations>();
-    manager.register_pass<ov::intel_gna::pass::RemoveInputConvert>();
-    manager.register_pass<ov::intel_gna::pass::RemoveOutputConvert>();
-    manager.register_pass<ov::pass::ConvertSequenceToTensorIterator>();
-    manager.register_pass<ov::pass::GRUCellDecomposition>();
-    manager.register_pass<ov::pass::LSTMCellDecomposition>();
-    manager.register_pass<ov::intel_gna::pass::ConvertDWSCToScaleShifts>();
-    manager.register_pass<ov::intel_gna::pass::ConvertPaddedToValidConv>();
-    manager.register_pass<ov::intel_gna::pass::Decompose2DConvTransposedWithBiasAF>(config.gnaPrecision);
-    manager.register_pass<ov::intel_gna::pass::Decompose2DConvTransposedWithBias>(config.gnaPrecision);
-    manager.register_pass<ov::intel_gna::pass::Decompose2DConv>(config.gnaPrecision);
+    std::vector<std::pair<std::string, std::shared_ptr<ov::pass::PassBase>>> transformations;
```

Comment:
It would be much cleaner if you extract the creation of transformations vector (all the transformations.emplace_back calls) into a separate method.